src/main/scala/io/citrine/lolo/trees/classification/ClassificationTree.scala

```diff
@@ -121,10 +123,10 @@ class ClassificationTrainingResult(
   * Classification tree
   */
 class ClassificationTree(
-    rootModelNode: ModelNode[PredictionResult[Char]],
+    rootModelNode: ModelNode[Char],
     inputEncoders: Seq[Option[CategoricalEncoder[Any]]],
     outputEncoder: CategoricalEncoder[Any]
-) extends Model[ClassificationResult] {
+) extends Model[Any] {
```

Comment:
This is one of the confusing parts I mentioned in the top level comments. A `ClassificationTree` is a `Learner[Any]` because it can train on any type of label data for classification. But the node structures that actually defines it is a bunch of `ModelNode[Char]` types because we encode the labels with a `CategoricalEncoder[Any]`. 

It works, but I can see why this is confusing.

Reply:
It makes sense to me, but I've spent a lot of time in this code lately

Reply:
I may encode this behavior a bit more firmly in one of the follow-up PRs. I've been thinking of making `Learner[T]` sealed and expliictly defining `RegressionLearner`, `ClassificationLearner`, etc..

I could imagine a scenario in which it makes sense to define `ClassificationLearner <: Learner[Any]` and `LeafClassifier <: Learner[Char]` as variants of the base trait.